diff --git a/nci/jni/NativeNfcManager.cpp b/nci/jni/NativeNfcManager.cpp
index d49997a6..f5030092 100644
--- a/nci/jni/NativeNfcManager.cpp
+++ b/nci/jni/NativeNfcManager.cpp
@@ -3758,6 +3758,20 @@ static void nfcManager_doFactoryReset(JNIEnv*, jobject) {
     return num;
   }
 
+  /*******************************************************************************
+  **
+  ** Function:       nfcManager_enableDebugNtf
+  **
+  ** Description:    Enable & disable the Lx debug notifications
+  **
+  ** Returns:        returns 0x00 in success case, 0x03 in failure case,
+  **                 0x01 is Nfc is off
+  *******************************************************************************/
+  static jint nfcManager_enableDebugNtf(JNIEnv * e, jobject o,
+                                        jbyte fieldValue) {
+    // QSSI.
+    return NFA_STATUS_OK;
+  }
   /*******************************************************************************
    **
    ** Function:        nfcManager_setPreferredSimSlot()
@@ -4601,6 +4615,7 @@ static void restartUiccListen(jint uiccSlot) {
     {"doGetSelectedUicc", "()I", (void*)nfcManager_doGetSelectedUicc},
     {"setPreferredSimSlot", "(I)I", (void*)nfcManager_setPreferredSimSlot},
     {"doNfcSelfTest", "(I)I", (void*)nfcManager_nfcSelfTest},
+    {"doEnableDebugNtf", "(B)I", (void*) nfcManager_enableDebugNtf},
 #endif
     {"doSetNfcSecure", "(Z)Z", (void*)nfcManager_doSetNfcSecure},
   };
