commit bd6757a0577dbcd1da04d0b5cac561937b9735cb
Author: nxf52135 <amit.sharma_1@nxp.com>
Date:   Wed Dec 1 12:11:40 2021 +0530

    [artf907077] Updated nfc driver name in the config files

diff --git a/SN100x/halimpl/conf/PN557/gen-config-files/libnfc-nxp-PN557_example.conf b/SN100x/halimpl/conf/PN557/gen-config-files/libnfc-nxp-PN557_example.conf
index aceb729..1028238 100755
--- a/SN100x/halimpl/conf/PN557/gen-config-files/libnfc-nxp-PN557_example.conf
+++ b/SN100x/halimpl/conf/PN557/gen-config-files/libnfc-nxp-PN557_example.conf
@@ -17,7 +17,7 @@ NXPLOG_TML_LOGLEVEL=0x03
 
 ###############################################################################
 # Nfc Device Node name
-NXP_NFC_DEV_NODE="/dev/pn553"
+NXP_NFC_DEV_NODE="/dev/nxp-nci"
 
 ###############################################################################
 # Extension for Mifare reader enable
@@ -395,4 +395,4 @@ NXP_DUAL_UICC_ENABLE=0x00
 # Enable            0x01
 NXP_DISCONNECT_TAG_IN_SCRN_OFF =0x00
 
-###############################################################################
\ No newline at end of file
+###############################################################################
diff --git a/SN100x/halimpl/conf/SN1xx/sn100/gen-config-files/libnfc-nxp-SN100U_example.conf b/SN100x/halimpl/conf/SN1xx/sn100/gen-config-files/libnfc-nxp-SN100U_example.conf
index 0c5d70d..4dd8f37 100644
--- a/SN100x/halimpl/conf/SN1xx/sn100/gen-config-files/libnfc-nxp-SN100U_example.conf
+++ b/SN100x/halimpl/conf/SN1xx/sn100/gen-config-files/libnfc-nxp-SN100U_example.conf
@@ -17,7 +17,7 @@ NFC_DEBUG_ENABLED=1
 
 ###############################################################################
 # Nfc Device Node name
-NXP_NFC_DEV_NODE="/dev/pn553"
+NXP_NFC_DEV_NODE="/dev/nxp-nci"
 
 ###############################################################################
 # Extension for Mifare reader enable
diff --git a/SN100x/halimpl/conf/SN1xx/sn110/gen-config-files/libnfc-nxp-SN100U_example.conf b/SN100x/halimpl/conf/SN1xx/sn110/gen-config-files/libnfc-nxp-SN100U_example.conf
index 20acc3f..9c0b5e6 100644
--- a/SN100x/halimpl/conf/SN1xx/sn110/gen-config-files/libnfc-nxp-SN100U_example.conf
+++ b/SN100x/halimpl/conf/SN1xx/sn110/gen-config-files/libnfc-nxp-SN100U_example.conf
@@ -17,7 +17,7 @@ NFC_DEBUG_ENABLED=1
 
 ###############################################################################
 # Nfc Device Node name
-NXP_NFC_DEV_NODE="/dev/pn553"
+NXP_NFC_DEV_NODE="/dev/nxp-nci"
 
 ###############################################################################
 # Extension for Mifare reader enable
diff --git a/SN100x/halimpl/conf/SN220/gen-config-files/libnfc-nxp-SN220U_example.conf b/SN100x/halimpl/conf/SN220/gen-config-files/libnfc-nxp-SN220U_example.conf
index 2d5f714..83b44b6 100644
--- a/SN100x/halimpl/conf/SN220/gen-config-files/libnfc-nxp-SN220U_example.conf
+++ b/SN100x/halimpl/conf/SN220/gen-config-files/libnfc-nxp-SN220U_example.conf
@@ -17,7 +17,7 @@ NFC_DEBUG_ENABLED=1
 
 ###############################################################################
 # Nfc Device Node name
-NXP_NFC_DEV_NODE="/dev/pn553"
+NXP_NFC_DEV_NODE="/dev/nxp-nci"
 
 ###############################################################################
 # Extension for Mifare reader enable
