diff --git a/drivers/nxp/nfc/Kbuild b/drivers/nxp/nfc/Kbuild
index 198d574..9a4e9a2 100644
--- a/drivers/nxp/nfc/Kbuild
+++ b/drivers/nxp/nfc/Kbuild
@@ -1,6 +1,6 @@
-obj-$(CONFIG_NXP_NFC_I2C) := pn553_i2c.o
+obj-$(CONFIG_NXP_NFC_I2C) := nfc_i2c.o
 
-pn553_i2c-y := common.o \
-               common_ese.o \
-               i2c_drv.o
+nfc_i2c-y := common.o \
+             common_ese.o \
+             i2c_drv.o
 
diff --git a/drivers/nxp/nfc/common.h b/drivers/nxp/nfc/common.h
index bcdd26d..675e06c 100644
--- a/drivers/nxp/nfc/common.h
+++ b/drivers/nxp/nfc/common.h
@@ -30,7 +30,7 @@
 #define CLASS_NAME			"nfc"
 
 /* NFC character device name, this will be in /dev/ */
-#define NFC_CHAR_DEV_NAME		"pn553"
+#define NFC_CHAR_DEV_NAME		"nxp-nci"
 
 /* NCI packet details */
 #define NCI_CMD				(0x20)
@@ -78,9 +78,9 @@
 #define ESE_SET_PWR			_IOW(NFC_MAGIC, 0x02, uint32_t)
 #define ESE_GET_PWR			_IOR(NFC_MAGIC, 0x03, uint32_t)
 
-#define DTS_IRQ_GPIO_STR		"nxp,pn544-irq"
-#define DTS_VEN_GPIO_STR		"nxp,pn544-ven"
-#define DTS_FWDN_GPIO_STR		"nxp,pn544-fw-dwnld"
+#define DTS_IRQ_GPIO_STR		"nxp,sn-irq"
+#define DTS_VEN_GPIO_STR		"nxp,sn-ven-rstn"
+#define DTS_FWDN_GPIO_STR		"nxp,sn-dwl-req"
 
 enum nfcc_ioctl_request {
 	/* NFC disable request with VEN LOW */
diff --git a/drivers/nxp/nfc/i2c_drv.h b/drivers/nxp/nfc/i2c_drv.h
index 1508fd4..628ca08 100644
--- a/drivers/nxp/nfc/i2c_drv.h
+++ b/drivers/nxp/nfc/i2c_drv.h
@@ -23,8 +23,8 @@
 #include <linux/i2c.h>
 
 /* kept same as dts */
-#define NFC_I2C_DRV_STR			"nxp,pn544"
-#define NFC_I2C_DEV_ID			"pn553"
+#define NFC_I2C_DRV_STR			"nxp,sn-nci"
+#define NFC_I2C_DEV_ID			"sn-i2c"
 
 /* Interface specific parameters */
 struct i2c_dev {
