diff --git a/hikey960.mk b/hikey960.mk
index 1b9799b..a1ba4dc 100644
--- a/hikey960.mk
+++ b/hikey960.mk
@@ -16,6 +16,7 @@ $(call inherit-product, $(SRC_TARGET_DIR)/product/core_64_bit.mk)
 $(call inherit-product, device/linaro/hikey/hikey960/device-hikey960.mk)
 $(call inherit-product, device/linaro/hikey/device-common.mk)
 $(call inherit-product, $(SRC_TARGET_DIR)/product/full_base.mk)
+$(call inherit-product, vendor/nxp/SN100x/device-nfc.mk)
 
 #
 # Overrides
diff --git a/hikey960/BoardConfig.mk b/hikey960/BoardConfig.mk
index 122fc67..30c6f57 100644
--- a/hikey960/BoardConfig.mk
+++ b/hikey960/BoardConfig.mk
@@ -1,4 +1,5 @@
 include device/linaro/hikey/BoardConfigCommon.mk
+include vendor/nxp/SN100x/BoardConfigNfc.mk
 
 TARGET_BOOTLOADER_BOARD_NAME := hikey960
 TARGET_BOARD_PLATFORM := hikey960
