diff --git a/lib/ext2fs/sparse_io.c b/lib/ext2fs/sparse_io.c
index 1b39b2e..d0828a8 100644
--- a/lib/ext2fs/sparse_io.c
+++ b/lib/ext2fs/sparse_io.c
@@ -67,7 +67,7 @@ static void free_sparse_blocks(struct sparse_map *sm)
 	sm->blocks = NULL;
 }
 
-static int sparse_import_segment(void *priv, const void *data, int len,
+static int sparse_import_segment(void *priv, const void *data, size_t len,
 				 unsigned int block, unsigned int nr_blocks)
 {
 	struct sparse_map *sm = priv;
