diff --git a/lib/libf2fs_io.c b/lib/libf2fs_io.c
index 4781517..1ed5ca5 100644
--- a/lib/libf2fs_io.c
+++ b/lib/libf2fs_io.c
@@ -114,7 +114,7 @@ static int sparse_write_blk(__u64 block, int count, const void *buf)
 	return 0;
 }
 
-static int sparse_import_segment(void *UNUSED(priv), const void *data, int len,
+static int sparse_import_segment(void *UNUSED(priv), const void *data, size_t len,
 		unsigned int block, unsigned int nr_blocks)
 {
 	/* Ignore chunk headers, only write the data */
