diff --git a/core/board_config.mk b/core/board_config.mk
index 2580a33..85f2be3 100644
--- a/core/board_config.mk
+++ b/core/board_config.mk
@@ -101,7 +101,7 @@ _board_strip_readonly_list += $(_build_broken_var_list)
 
 # Conditional to building on linux, as dex2oat currently does not work on darwin.
 ifeq ($(HOST_OS),linux)
-  WITH_DEXPREOPT := true
+  WITH_DEXPREOPT := false
 endif
 
 # ###############################################################
diff --git a/core/dex_preopt_config.mk b/core/dex_preopt_config.mk
index a494991..af34346 100644
--- a/core/dex_preopt_config.mk
+++ b/core/dex_preopt_config.mk
@@ -34,13 +34,13 @@ ifeq ($(HOST_OS),linux)
 
   # Non eng linux builds must have preopt enabled so that system server doesn't run as interpreter
   # only. b/74209329
-  ifeq (,$(filter eng, $(TARGET_BUILD_VARIANT)))
-    ifneq (true,$(WITH_DEXPREOPT))
-      ifneq (true,$(WITH_DEXPREOPT_BOOT_IMG_AND_SYSTEM_SERVER_ONLY))
-        $(call pretty-error, DEXPREOPT must be enabled for user and userdebug builds)
-      endif
-    endif
-  endif
+#  ifeq (,$(filter eng, $(TARGET_BUILD_VARIANT)))
+#    ifneq (true,$(WITH_DEXPREOPT))
+#      ifneq (true,$(WITH_DEXPREOPT_BOOT_IMG_AND_SYSTEM_SERVER_ONLY))
+#        $(call pretty-error, DEXPREOPT must be enabled for user and userdebug builds)
+#      endif
+#    endif
+#  endif
 endif
 
 # Default to debug version to help find bugs.
diff --git a/core/tasks/check_boot_jars/package_whitelist.txt b/core/tasks/check_boot_jars/package_whitelist.txt
index b0becba..493832c 100644
--- a/core/tasks/check_boot_jars/package_whitelist.txt
+++ b/core/tasks/check_boot_jars/package_whitelist.txt
@@ -238,6 +238,10 @@ org\.apache\.xalan\.xslt
 com\.google\.android\..*
 com\.google\.vr\.platform.*
 
+##########Classes for NXP NFC ######
+com\.nxp\..*
+com\.gsma\.services\..*
+
 ###################################################
 # Packages used for Android in Chrome OS
 org\.chromium\.arc
diff --git a/target/product/gsi/29.txt b/target/product/gsi/29.txt
index 14faba5..b05d4f3
--- a/target/product/gsi/29.txt
+++ b/target/product/gsi/29.txt
@@ -159,6 +159,7 @@ VNDK-core: android.hardware.radio@1.3.so
 VNDK-core: android.hardware.radio@1.4.so
 VNDK-core: android.hardware.secure_element@1.0.so
 VNDK-core: android.hardware.secure_element@1.1.so
+VNDK-core: android.hardware.secure_element@1.2.so
 VNDK-core: android.hardware.sensors@1.0.so
 VNDK-core: android.hardware.sensors@2.0.so
 VNDK-core: android.hardware.soundtrigger@2.0.so

