diff --git a/hikey960.mk b/hikey960.mk
index 2e338e9..033049d 100644
--- a/hikey960.mk
+++ b/hikey960.mk
@@ -11,7 +11,7 @@ else
   ifeq ($(TARGET_KERNEL_USE), 4.9)
     HIKEY_USE_DRM_HWCOMPOSER := false
   else
-    HIKEY_USE_DRM_HWCOMPOSER := true
+    HIKEY_USE_DRM_HWCOMPOSER := false
   endif
   HIKEY_USE_LEGACY_TI_BLUETOOTH := false
 endif
@@ -22,6 +22,7 @@ $(call inherit-product, $(SRC_TARGET_DIR)/product/core_64_bit.mk)
 $(call inherit-product, device/linaro/hikey/hikey960/device-hikey960.mk)
 $(call inherit-product, device/linaro/hikey/device-common.mk)
 $(call inherit-product, $(SRC_TARGET_DIR)/product/full_base.mk)
+$(call inherit-product, vendor/nxp/SN100x/device-nfc.mk)
 
 #setup dm-verity configs
 PRODUCT_SYSTEM_VERITY_PARTITION := /dev/block/platform/soc/ff3b0000.ufs/by-name/system
@@ -31,6 +32,8 @@ PRODUCT_SUPPORTS_BOOT_SIGNER := false
 PRODUCT_SUPPORTS_VERITY_FEC := false
 
 PRODUCT_PROPERTY_OVERRIDES += ro.opengles.version=196608
+PRODUCT_PROPERTY_OVERRIDES += ro.product.first_api_level=28
+PRODUCT_PROPERTY_OVERRIDES += ro.config.low_ram=false
 
 #
 # Overrides
diff --git a/hikey960/BoardConfig.mk b/hikey960/BoardConfig.mk
index 3cd7a63..e2078bf 100644
--- a/hikey960/BoardConfig.mk
+++ b/hikey960/BoardConfig.mk
@@ -1,4 +1,5 @@
 include device/linaro/hikey/BoardConfigCommon.mk
+include vendor/nxp/SN100x/BoardConfigNfc.mk
 
 TARGET_BOOTLOADER_BOARD_NAME := hikey960
 TARGET_BOARD_PLATFORM := hikey960
