diff --git a/core/java/android/os/BinderProxy.java b/core/java/android/os/BinderProxy.java
index 3d466a0b..0c4aa15d 100644
--- a/core/java/android/os/BinderProxy.java
+++ b/core/java/android/os/BinderProxy.java
@@ -218,7 +218,7 @@ public final class BinderProxy implements IBinder {
                 Log.v(Binder.TAG, "BinderProxy map growth! bucket size = " + size
                         + " total = " + totalSize);
                 mWarnBucketSize += WARN_INCREMENT;
-                if (Build.IS_DEBUGGABLE && totalSize >= CRASH_AT_SIZE) {
+                /*if (Build.IS_DEBUGGABLE && totalSize >= CRASH_AT_SIZE) {
                     // Use the number of uncleared entries to determine whether we should
                     // really report a histogram and crash. We don't want to fundamentally
                     // change behavior for a debuggable process, so we GC only if we are
@@ -236,7 +236,7 @@ public final class BinderProxy implements IBinder {
                                 + (totalSize - totalUnclearedSize) + " of " + totalSize
                                 + " are cleared");
                     }
-                }
+                }*/
             }
         }
 
diff --git a/services/core/java/com/android/server/Watchdog.java b/services/core/java/com/android/server/Watchdog.java
index fcd049f1..0556cec1 100644
--- a/services/core/java/com/android/server/Watchdog.java
+++ b/services/core/java/com/android/server/Watchdog.java
@@ -158,7 +158,7 @@ public class Watchdog {
     private ActivityManagerService mActivity;
 
     private IActivityController mController;
-    private boolean mAllowRestart = true;
+    private boolean mAllowRestart = false;
     private final List<Integer> mInterestingJavaPids = new ArrayList<>();
 
     private final TraceErrorLogger mTraceErrorLogger;
@@ -432,7 +432,7 @@ public class Watchdog {
 
     public void setAllowRestart(boolean allowRestart) {
         synchronized (mLock) {
-            mAllowRestart = allowRestart;
+            mAllowRestart = false;
         }
     }
 
