diff --git a/arch/arm64/boot/dts/hisilicon/hi3660-hikey960.dts b/arch/arm64/boot/dts/hisilicon/hi3660-hikey960.dts
index 5c8cbd73..c981818 100644
--- a/arch/arm64/boot/dts/hisilicon/hi3660-hikey960.dts
+++ b/arch/arm64/boot/dts/hisilicon/hi3660-hikey960.dts
@@ -202,8 +202,9 @@
 
 			ldo11: LDO11 { /* Low Speed Connector */
 				regulator-name = "VOUT11_1V8_2V95";
-				regulator-min-microvolt = <1750000>;
+				regulator-min-microvolt = <1825000>;
 				regulator-max-microvolt = <3300000>;
+                                regulator-always-on;
 				regulator-enable-ramp-delay = <240>;
 			};
 
diff --git a/arch/arm64/boot/dts/hisilicon/hi3660.dtsi b/arch/arm64/boot/dts/hisilicon/hi3660.dtsi
index 80334e5..f88b37a8 100644
--- a/arch/arm64/boot/dts/hisilicon/hi3660.dtsi
+++ b/arch/arm64/boot/dts/hisilicon/hi3660.dtsi
@@ -475,12 +475,22 @@
 			interrupts = <GIC_SPI 314 IRQ_TYPE_LEVEL_HIGH>;
 			#address-cells = <1>;
 			#size-cells = <0>;
-			clock-frequency = <400000>;
+			clock-frequency = <1000000>;
 			clocks = <&crg_ctrl HI3660_CLK_GATE_I2C7>;
 			resets = <&crg_rst 0x60 14>;
 			pinctrl-names = "default";
 			pinctrl-0 = <&i2c7_pmx_func &i2c7_cfg_func>;
-			status = "disabled";
+			status = "ok";
+			pn544: pn544@28 {
+				compatible = "nxp,pn544";
+				reg = <0x28>;
+				//As per Hi960 mapping ex:GPIO208
+				nxp,pn544-irq = <&gpio26 0 0>;
+				nxp,pn544-ven = <&gpio26 1 0>;
+				nxp,pn544-fw-dwnld = <&gpio26 2 0>;
+				nxp,pn544-iso-pwr-rst = <&gpio6 4 0>;
+				nxp,pn544-ese-pwr = <&gpio26 4 0>;
+			};
 		};
 
 		uart0: serial@fdf02000 {
@@ -1000,7 +1010,16 @@
 			pinctrl-0 = <&spi2_pmx_func>;
 			num-cs = <1>;
 			cs-gpios = <&gpio27 2 0>;
-			status = "disabled";
+			status = "ok";
+			p61@0 {
+				compatible = "nxp,p61";
+				reg = <0>;
+				nxp,p61-irq = <&gpio2 3 0>;
+				nxp,p61-rst = <&gpio2 5 0>;
+				nxp,trusted-se = <&gpio26 4 0>;
+				spi-max-frequency = <20000000>;
+				nxp,nfcc = "2-0028";
+			};
 		};
 
 		spi3: spi@ff3b3000 {
diff --git a/drivers/Kconfig b/drivers/Kconfig
index ef7084b..ac2a2d1 100644
--- a/drivers/Kconfig
+++ b/drivers/Kconfig
@@ -222,4 +222,6 @@ source "drivers/siox/Kconfig"
 source "drivers/slimbus/Kconfig"
 
 source "drivers/energy_model/Kconfig"
+
+source "drivers/nxp/Kconfig"
 endmenu
diff --git a/drivers/Makefile b/drivers/Makefile
index 1e847e2..b65e1f5 100644
--- a/drivers/Makefile
+++ b/drivers/Makefile
@@ -69,7 +69,7 @@ obj-$(CONFIG_FB_INTEL)          += video/fbdev/intelfb/
 
 obj-$(CONFIG_PARPORT)		+= parport/
 obj-$(CONFIG_NVM)		+= lightnvm/
-obj-y				+= base/ block/ misc/ mfd/ nfc/
+obj-y				+= base/ block/ misc/ mfd/
 obj-$(CONFIG_LIBNVDIMM)		+= nvdimm/
 obj-$(CONFIG_DAX)		+= dax/
 obj-$(CONFIG_DMA_SHARED_BUFFER) += dma-buf/
@@ -188,3 +188,4 @@ obj-$(CONFIG_MULTIPLEXER)	+= mux/
 obj-$(CONFIG_UNISYS_VISORBUS)	+= visorbus/
 obj-$(CONFIG_SIOX)		+= siox/
 obj-$(CONFIG_GNSS)		+= gnss/
+obj-$(CONFIG_NXP_NFC)            += nxp/
diff --git a/drivers/usb/dwc3/core.c b/drivers/usb/dwc3/core.c
index 68c9c91..1c90d9b 100644
--- a/drivers/usb/dwc3/core.c
+++ b/drivers/usb/dwc3/core.c
@@ -1814,8 +1814,7 @@ static int dwc3_runtime_idle(struct device *dev)
 }
 #endif /* CONFIG_PM */
 
-#ifdef CONFIG_PM_SLEEP
-static int dwc3_suspend(struct device *dev)
+static int __maybe_unused dwc3_suspend(struct device *dev)
 {
 	struct dwc3	*dwc = dev_get_drvdata(dev);
 	int		ret;
@@ -1829,7 +1828,7 @@ static int dwc3_suspend(struct device *dev)
 	return 0;
 }
 
-static int dwc3_resume(struct device *dev)
+static int __maybe_unused dwc3_resume(struct device *dev)
 {
 	struct dwc3	*dwc = dev_get_drvdata(dev);
 	int		ret;
@@ -1847,7 +1846,7 @@ static int dwc3_resume(struct device *dev)
 	return 0;
 }
 
-static void dwc3_complete(struct device *dev)
+static void  __maybe_unused dwc3_complete(struct device *dev)
 {
 	struct dwc3	*dwc = dev_get_drvdata(dev);
 	u32		reg;
@@ -1860,11 +1859,12 @@ static void dwc3_complete(struct device *dev)
 		dwc3_writel(dwc->regs, DWC3_GUCTL3, reg);
 	}
 }
-#endif /* CONFIG_PM_SLEEP */
 
 static const struct dev_pm_ops dwc3_dev_pm_ops = {
 	SET_SYSTEM_SLEEP_PM_OPS(dwc3_suspend, dwc3_resume)
+#ifdef CONFIG_PM_SLEEP
 	.complete = dwc3_complete,
+#endif /* CONFIG_PM_SLEEP */
 	SET_RUNTIME_PM_OPS(dwc3_runtime_suspend, dwc3_runtime_resume,
 			dwc3_runtime_idle)
 };
diff --git a/kernel/irq/manage.c b/kernel/irq/manage.c
index 23bcfa7..fc59c18 100644
--- a/kernel/irq/manage.c
+++ b/kernel/irq/manage.c
@@ -685,7 +685,7 @@ int irq_set_irq_wake(unsigned int irq, unsigned int on)
 		}
 	} else {
 		if (desc->wake_depth == 0) {
-			WARN(1, "Unbalanced IRQ %d wake disable\n", irq);
+			WARN(0, "Unbalanced IRQ %d wake disable\n", irq);
 		} else if (--desc->wake_depth == 0) {
 			ret = set_irq_wake_real(irq, on);
 			if (ret)
