diff --git a/arch/arm64/boot/dts/hisilicon/hi3660-hikey960.dts b/arch/arm64/boot/dts/hisilicon/hi3660-hikey960.dts
index 5c8cbd7382e2..c981818643cf 100644
--- a/arch/arm64/boot/dts/hisilicon/hi3660-hikey960.dts
+++ b/arch/arm64/boot/dts/hisilicon/hi3660-hikey960.dts
@@ -202,8 +202,9 @@
 
 			ldo11: LDO11 { /* Low Speed Connector */
 				regulator-name = "VOUT11_1V8_2V95";
-				regulator-min-microvolt = <1750000>;
+				regulator-min-microvolt = <1825000>;
 				regulator-max-microvolt = <3300000>;
+                                regulator-always-on;
 				regulator-enable-ramp-delay = <240>;
 			};
 
diff --git a/arch/arm64/boot/dts/hisilicon/hi3660.dtsi b/arch/arm64/boot/dts/hisilicon/hi3660.dtsi
index 80334e50064d..cbb0e050012a 100644
--- a/arch/arm64/boot/dts/hisilicon/hi3660.dtsi
+++ b/arch/arm64/boot/dts/hisilicon/hi3660.dtsi
@@ -475,12 +475,22 @@
 			interrupts = <GIC_SPI 314 IRQ_TYPE_LEVEL_HIGH>;
 			#address-cells = <1>;
 			#size-cells = <0>;
-			clock-frequency = <400000>;
+			clock-frequency = <1000000>;
 			clocks = <&crg_ctrl HI3660_CLK_GATE_I2C7>;
 			resets = <&crg_rst 0x60 14>;
 			pinctrl-names = "default";
 			pinctrl-0 = <&i2c7_pmx_func &i2c7_cfg_func>;
-			status = "disabled";
+			status = "ok";
+			pn544: pn544@28 {
+				compatible = "nxp,pn544";
+				reg = <0x28>;
+				//As per Hi960 mapping ex:GPIO208
+				nxp,pn544-irq = <&gpio26 0 0>;
+				nxp,pn544-ven = <&gpio26 1 0>;
+                                nxp,pn544-fw-dwnld = <&gpio26 2 0>;
+				nxp,pn544-iso-pwr-rst = <&gpio6 4 0>;
+				nxp,pn544-ese-pwr = <&gpio26 4 0>;
+			};
 		};
 
 		uart0: serial@fdf02000 {
@@ -997,10 +1007,19 @@
 			clocks = <&crg_ctrl HI3660_CLK_GATE_SPI2>;
 			clock-names = "apb_pclk";
 			pinctrl-names = "default";
-			pinctrl-0 = <&spi2_pmx_func>;
+			pinctrl-0 = <&spi2_pmx_func &spi2_cfg_func>;
 			num-cs = <1>;
 			cs-gpios = <&gpio27 2 0>;
-			status = "disabled";
+			status = "ok";
+			p61@0 {
+				compatible = "nxp,p61";
+				reg = <0>;
+				nxp,p61-irq = <&gpio2 3 0>;
+				nxp,p61-rst = <&gpio2 5 0>;
+				nxp,trusted-se = <&gpio26 4 0>;
+				spi-max-frequency = <20000000>;
+				nxp,nfcc = "2-0028";
+			};
 		};
 
 		spi3: spi@ff3b3000 {
diff --git a/arch/arm64/boot/dts/hisilicon/hikey960-pinctrl.dtsi b/arch/arm64/boot/dts/hisilicon/hikey960-pinctrl.dtsi
index d11efc81958c..2980adced30f 100644
--- a/arch/arm64/boot/dts/hisilicon/hikey960-pinctrl.dtsi
+++ b/arch/arm64/boot/dts/hisilicon/hikey960-pinctrl.dtsi
@@ -1031,7 +1031,7 @@
 					PULL_UP
 				>;
 				pinctrl-single,drive-strength = <
-					DRIVE7_02MA DRIVE6_MASK
+					DRIVE7_04MA DRIVE6_MASK
 				>;
 			};
 
diff --git a/arch/arm64/configs/hikey960_defconfig b/arch/arm64/configs/hikey960_defconfig
index 52c5d51d2e8b..4dc7b31fc564 100644
--- a/arch/arm64/configs/hikey960_defconfig
+++ b/arch/arm64/configs/hikey960_defconfig
@@ -596,3 +596,8 @@ CONFIG_CORESIGHT=y
 CONFIG_CORESIGHT_LINK_AND_SINK_TMC=y
 CONFIG_CORESIGHT_SINK_TPIU=y
 CONFIG_CORESIGHT_SOURCE_ETM4X=y
+CONFIG_NXP_NFC=y
+CONFIG_NXP_PN80T=y
+CONFIG_NFC_PN553_DEVICES=y
+CONFIG_NXP_P73_DEVICES=y
+CONFIG_SUSPEND=n
diff --git a/drivers/Kconfig b/drivers/Kconfig
index ff09d476655a..f33a44d17813 100644
--- a/drivers/Kconfig
+++ b/drivers/Kconfig
@@ -224,4 +224,6 @@ source "drivers/siox/Kconfig"
 source "drivers/slimbus/Kconfig"
 
 source "drivers/energy_model/Kconfig"
+
+source "drivers/nxp/Kconfig"
 endmenu
diff --git a/drivers/Makefile b/drivers/Makefile
index 1e847e29661c..ee005942f2b5 100644
--- a/drivers/Makefile
+++ b/drivers/Makefile
@@ -69,7 +69,7 @@ obj-$(CONFIG_FB_INTEL)          += video/fbdev/intelfb/
 
 obj-$(CONFIG_PARPORT)		+= parport/
 obj-$(CONFIG_NVM)		+= lightnvm/
-obj-y				+= base/ block/ misc/ mfd/ nfc/
+obj-y				+= base/ block/ misc/ mfd/
 obj-$(CONFIG_LIBNVDIMM)		+= nvdimm/
 obj-$(CONFIG_DAX)		+= dax/
 obj-$(CONFIG_DMA_SHARED_BUFFER) += dma-buf/
@@ -188,3 +188,4 @@ obj-$(CONFIG_MULTIPLEXER)	+= mux/
 obj-$(CONFIG_UNISYS_VISORBUS)	+= visorbus/
 obj-$(CONFIG_SIOX)		+= siox/
 obj-$(CONFIG_GNSS)		+= gnss/
+obj-y			        += nxp/
diff --git a/drivers/i2c/busses/i2c-designware-platdrv.c b/drivers/i2c/busses/i2c-designware-platdrv.c
index b5750fd85125..ae996853b475 100644
--- a/drivers/i2c/busses/i2c-designware-platdrv.c
+++ b/drivers/i2c/busses/i2c-designware-platdrv.c
@@ -353,6 +353,64 @@ static int dw_i2c_plat_probe(struct platform_device *pdev)
 
 	pm_runtime_enable(&pdev->dev);
 
+       /* Manually configuring I2C controller "high-level and low-level duration (in tick count" count */
+       switch (t->bus_freq_hz) {
+       case 1000000:
+       /*  target:1Mhz Frequency with i2c0 controller (32Mhz oscillator):
+               cnt = 32Mhz / 1Mhz = 32
+               hcnt + lcnt = 32
+               hcnt ~= lnct = 32/2 ~= 16
+               hcnt = hcnt - Controller specific adjustment
+               lcnt = lcnt - Controller Specific adjustment
+               dev->fp_hcnt = 10;
+			   dev->fp_lcnt =  8; */
+
+       /*  target:1Mhz Frequency with i2c1 controller (100Mhz oscillator):
+        cnt = 100Mhz / 1Mhz = 100
+        hcnt + lcnt = 100
+        hcnt ~= lnct = 100/2 ~= 50
+               hcnt = hcnt - Controller specific adjustment
+               lcnt = lcnt - Controller Specific adjustment
+               dev->fp_hcnt = 50;
+        dev->fp_lcnt =  50; */
+        dev->fp_hcnt = 45;
+        dev->fp_lcnt = 43;
+        dev_err(&pdev->dev,
+            "##### Fast mode plus set: freq = %d", t->bus_freq_hz);
+               break;
+       case 3400000:
+               /*  target:3.4Mhz Frequency with i2c0 controller (32Mhz oscillator):
+               Did not work */
+
+               /*  target:3.4Mhz Frequency with i2c1 controller (100Mhz oscillator):
+               cnt = 106Mhz / 3.4Mhz = ~30
+               hcnt + lcnt = ~30
+               hcnt ~= lnct = 3/2 ~= 15
+               hcnt = hcnt - Controller specific adjustment
+               lcnt = lcnt - Controller Specific adjustment
+               */
+        dev->hs_hcnt = 8;
+        dev->hs_lcnt = 8;
+        dev_err(&pdev->dev,
+           "##### High speed mode set: freq = %d", t->bus_freq_hz);
+               break;
+       case 400000:
+               /*  target:400KHz Frequency with i2c1 controller (100Mhz oscillator):
+               cnt = 100Mhz / .4Mhz = 250
+               hcnt + lcnt = 250
+               hcnt ~= lnct = 250/2 ~= 125
+               hcnt = hcnt - Controller specific adjustment
+               lcnt = lcnt - Controller Specific adjustment
+               */
+               dev->fs_hcnt = 125;
+               dev->fs_lcnt = 123;
+               break;
+       default:
+               dev_err(&pdev->dev,
+            "##### Standard mode set: freq = %d", t->bus_freq_hz);
+               break;
+       }
+
 	if (dev->mode == DW_IC_SLAVE)
 		ret = i2c_dw_probe_slave(dev);
 	else
diff --git a/drivers/usb/dwc3/core.c b/drivers/usb/dwc3/core.c
index 5e33fd4f7a7d..f5c9130747ba 100644
--- a/drivers/usb/dwc3/core.c
+++ b/drivers/usb/dwc3/core.c
@@ -1822,8 +1822,7 @@ static int dwc3_runtime_idle(struct device *dev)
 }
 #endif /* CONFIG_PM */
 
-#ifdef CONFIG_PM_SLEEP
-static int dwc3_suspend(struct device *dev)
+static int __maybe_unused dwc3_suspend(struct device *dev)
 {
 	struct dwc3	*dwc = dev_get_drvdata(dev);
 	int		ret;
@@ -1837,7 +1836,7 @@ static int dwc3_suspend(struct device *dev)
 	return 0;
 }
 
-static int dwc3_resume(struct device *dev)
+static int __maybe_unused dwc3_resume(struct device *dev)
 {
 	struct dwc3	*dwc = dev_get_drvdata(dev);
 	int		ret;
@@ -1855,7 +1854,7 @@ static int dwc3_resume(struct device *dev)
 	return 0;
 }
 
-static void dwc3_complete(struct device *dev)
+static void  __maybe_unused dwc3_complete(struct device *dev)
 {
 	struct dwc3	*dwc = dev_get_drvdata(dev);
 	u32		reg;
@@ -1868,11 +1867,12 @@ static void dwc3_complete(struct device *dev)
 		dwc3_writel(dwc->regs, DWC3_GUCTL3, reg);
 	}
 }
-#endif /* CONFIG_PM_SLEEP */
 
 static const struct dev_pm_ops dwc3_dev_pm_ops = {
 	SET_SYSTEM_SLEEP_PM_OPS(dwc3_suspend, dwc3_resume)
+#ifdef CONFIG_PM_SLEEP
 	.complete = dwc3_complete,
+#endif /* CONFIG_PM_SLEEP */
 	SET_RUNTIME_PM_OPS(dwc3_runtime_suspend, dwc3_runtime_resume,
 			dwc3_runtime_idle)
 };
diff --git a/kernel/irq/manage.c b/kernel/irq/manage.c
index ad295ba00b0d..c1f183e3cc81 100644
--- a/kernel/irq/manage.c
+++ b/kernel/irq/manage.c
@@ -679,7 +679,7 @@ int irq_set_irq_wake(unsigned int irq, unsigned int on)
 		}
 	} else {
 		if (desc->wake_depth == 0) {
-			WARN(1, "Unbalanced IRQ %d wake disable\n", irq);
+			WARN(0, "Unbalanced IRQ %d wake disable\n", irq);
 		} else if (--desc->wake_depth == 0) {
 			ret = set_irq_wake_real(irq, on);
 			if (ret)
