diff --git a/extns/impl/Nxp_Features.h b/extns/impl/Nxp_Features.h
index 7d3009f..e7cea36 100644
--- a/extns/impl/Nxp_Features.h
+++ b/extns/impl/Nxp_Features.h
@@ -376,7 +376,7 @@ extern tNfc_featureList nfcFL;
             \
             \
             nfcFL.eseFL._ESE_ETSI12_PROP_INIT = true;                       \
-            nfcFL.eseFL._EXCLUDE_NV_MEM_DEPENDENCY = false;                 \
+            nfcFL.eseFL._EXCLUDE_NV_MEM_DEPENDENCY = true;                 \
             \
             \
             nfcFL.platformFL._NFCC_RESET_RSP_LEN = 0x10U;                   \
diff --git a/halimpl/hal/phNxpNciHal.c b/halimpl/hal/phNxpNciHal.c
index 2669c87..a916889 100644
--- a/halimpl/hal/phNxpNciHal.c
+++ b/halimpl/hal/phNxpNciHal.c
@@ -58,7 +58,6 @@ static uint8_t fw_download_success = 0;
 static uint8_t config_access = false;
 static uint8_t config_success = true;
 static NFCSTATUS phNxpNciHal_FwDwnld(uint16_t aType);
-static void phNxpNciHal_check_delete_nfaStorage_DHArea();
 /* NCI HAL Control structure */
 phNxpNciHal_Control_t nxpncihal_ctrl;
 
@@ -1089,29 +1088,6 @@ void read_retry() {
   }
 }
 /*******************************************************************************
-**
-** Function         phNxpNciHal_check_delete_nfaStorage_DHArea
-**
-** Description      check the file and delete if present.
-**
-**
-** Returns          void
-**
-*******************************************************************************/
-void phNxpNciHal_check_delete_nfaStorage_DHArea() {
-  struct stat st;
-  int ret = 0;
-  const char config_eseinfo_path[] = "/data/nfc/nfaStorage.bin1";
-  if (stat(config_eseinfo_path, &st) == -1) {
-    ALOGD("%s: file not found %s", __func__, config_eseinfo_path);
-  } else {
-    ALOGD("%s: file found %s, delete it", __func__, config_eseinfo_path);
-    ret = remove(config_eseinfo_path);
-    if(ret!=0)
-      ALOGE("%s: error deleting file %s", __func__, config_eseinfo_path);
-  }
-}
-/*******************************************************************************
  **
  ** Function:        phNxpNciHal_lastResetNtfReason()
  **
@@ -1371,9 +1347,6 @@ int phNxpNciHal_core_initialized(uint8_t* p_core_init_rsp_params) {
   NXPLOG_NCIHAL_D("EEPROM_fw_dwnld_flag : 0x%02x SetConfigAlways flag : 0x%02x",
                   fw_dwnld_flag, setConfigAlways);
 
-  if (fw_download_success == 0x01) {
-    phNxpNciHal_check_delete_nfaStorage_DHArea();
-  }
   }
   if ((true == fw_dwnld_flag) || (true == setConfigAlways) ||
       isNxpRFConfigModified() || isNxpConfigModified()) {
@@ -2800,10 +2773,7 @@ void phNxpNciHal_do_factory_reset(void) {
  *
  ******************************************************************************/
 static void phNxpNciHal_check_factory_reset(void) {
-  struct stat st;
-  int ret = 0;
   NFCSTATUS status = NFCSTATUS_FAILED;
-  const char config_eseinfo_path[] = "/data/nfc/nfaStorage.bin1";
   uint8_t *reset_ese_session_identity_set;
   uint8_t ese_session_dyn_uicc_nv[] = {
             0x20, 0x02, 0x17, 0x02,0xA0, 0xEA, 0x08, 0xFF, 0xFF, 0xFF,
@@ -2842,14 +2812,7 @@ static void phNxpNciHal_check_factory_reset(void) {
       len = sizeof(ese_session);
     }
   }
-  if (stat(config_eseinfo_path, &st) == -1) {
-    NXPLOG_NCIHAL_D("%s file not present = %s", __func__, config_eseinfo_path);
-    ret = -1;
-  } else {
-    ret = 0;
-  }
 
-  if (ret == -1) {
     if((nfcFL.nfccFL._NFC_NXP_STAT_DUAL_UICC_WO_EXT_SWITCH) ||
             (nfcFL.nfccFL._NFCC_DYNAMIC_DUAL_UICC)) {
         status = phNxpNciHal_send_ext_cmd(len, reset_ese_session_identity_set);
@@ -2857,7 +2820,6 @@ static void phNxpNciHal_check_factory_reset(void) {
     if (status != NFCSTATUS_SUCCESS) {
       NXPLOG_NCIHAL_E("NXP reset_ese_session_identity_set command failed");
     }
-  }
 }
 
 /******************************************************************************
