diff --git a/hardware/nxp/nfc/snxxx/libnfc_vendor_extn/inc/common/NfcExtensionConstants.h b/hardware/nxp/nfc/snxxx/libnfc_vendor_extn/inc/common/NfcExtensionConstants.h
index a75e6e37..a897d633 100644
--- a/hardware/nxp/nfc/snxxx/libnfc_vendor_extn/inc/common/NfcExtensionConstants.h
+++ b/hardware/nxp/nfc/snxxx/libnfc_vendor_extn/inc/common/NfcExtensionConstants.h
@@ -69,6 +69,8 @@ constexpr uint8_t EXT_NFC_STATUS_INDEX = 0x03;
 constexpr uint8_t EXT_NFC_STATUS_OK = 0x00;
 constexpr uint8_t EXT_NFC_STATUS_SEMANTIC_ERROR = 0x06;
 
+constexpr uint8_t NCI_MT_DATA_UNCHAINED = 0x00;
+constexpr uint8_t NCI_MT_DATA_CHAINED = 0x10;
 constexpr uint8_t NCI_MT_CMD = 0x20;
 constexpr uint8_t NCI_MT_RSP = 0x40;
 constexpr uint8_t NCI_MT_NTF = 0x60;
diff --git a/hardware/nxp/nfc/snxxx/libnfc_vendor_extn/src/features/nci_state_monitor/NciStateMonitor.cpp b/hardware/nxp/nfc/snxxx/libnfc_vendor_extn/src/features/nci_state_monitor/NciStateMonitor.cpp
index 93a7f988..8b904b4f 100644
--- a/hardware/nxp/nfc/snxxx/libnfc_vendor_extn/src/features/nci_state_monitor/NciStateMonitor.cpp
+++ b/hardware/nxp/nfc/snxxx/libnfc_vendor_extn/src/features/nci_state_monitor/NciStateMonitor.cpp
@@ -152,6 +152,12 @@ NFCSTATUS NciStateMonitor::handleVendorNciRspNtf(uint16_t dataLen,
 NFCSTATUS NciStateMonitor::processNciCmd(uint16_t dataLen,
                                          const uint8_t *pData) {
   NXPLOG_EXTNS_D(NXPLOG_ITEM_NXP_GEN_EXTN, "NciStateMonitor %s Enter", __func__);
+  if (pData[0] == NCI_MT_DATA_UNCHAINED || pData[0] == NCI_MT_DATA_CHAINED) {
+    NXPLOG_EXTNS_E(NXPLOG_ITEM_NXP_GEN_EXTN,
+                   "NciStateMonitor %s, data packet received, skip processing",
+                   __func__);
+    return NFCSTATUS_EXTN_FEATURE_FAILURE;
+  }
   uint8_t num = 0;
   constexpr uint16_t NCI_EE_MODE_SET_CMD_GID_OID =
       (((NCI_MT_CMD | NCI_GID_EE_MANAGE) << 8) | NCI_EE_MODE_SET_OID);
